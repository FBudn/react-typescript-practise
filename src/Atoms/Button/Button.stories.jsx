import React from "react";
import Button from "./Button";

export default {
  title: "Atoms/Button",
  component: Button,
};

const Template = (args) => <Button {...args} />;

export const FlexFullWidthLOGIN = Template.bind({});
FlexFullWidthLOGIN.args = {
  children: "LOGIN",
};

export const FlexFullWidthSIGNUP = Template.bind({});
FlexFullWidthSIGNUP.args = {
  children: "SIGN UP",
};

/*

export const FlexStyleTesting = Template.bind({});
FlexStyleTesting.args = {
  children: "I'm for testing",
  storybookTesting: { backgroundColor: "green", width: "50%" },
};
*/
/*
Stories trzymam przy komponentach po prostu
decorator pod konkretny template ogarnąć czy się da i zrobić 

*/
