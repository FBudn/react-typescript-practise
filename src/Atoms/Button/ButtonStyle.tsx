import styled from "styled-components";

export const ButtonContainer = styled.button`
  display: flex;
  width: 100%;
  height: 40px;
  border-radius: 5px;
  border: none;
  justify-content: center;
  align-items: center;
  font-weight: bold;
  color: white;
  cursor: pointer;
  background: rgb(236 72 153);
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);
  --tw-shadow-color: #94a3b8;
  margin: 6px 0px 6px 0px;
`;
