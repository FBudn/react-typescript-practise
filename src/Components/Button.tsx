import React from "react";
import { ButtonContainer } from "./Styles";

export interface ButtonProps {
  text: string;
  onClick: () => void;
  // checkboxValue: any;
}

const Button: React.FC<ButtonProps> = ({ text, onClick }) => {
  return <ButtonContainer onClick={onClick}>{text}</ButtonContainer>;
};

export default Button;

// onClick={() => handleValues}
// tu w funkcji ktora przekazuje do Buttona mogę sobie spiąć w obiekt stany email itd zeby wyswietlic w 1
