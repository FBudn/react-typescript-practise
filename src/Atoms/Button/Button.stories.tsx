import { Meta, Story } from "@storybook/react";
import React from "react";
import Button, { ButtonProps } from "./Button";

export default {
  title: "Atoms/Button",
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const FlexFullWidthLOGIN = Template.bind({});
FlexFullWidthLOGIN.args = {
  children: "LOGIN",
};

export const FlexFullWidthSIGNUP = Template.bind({});
FlexFullWidthSIGNUP.args = {
  children: "SIGN UP",
};
