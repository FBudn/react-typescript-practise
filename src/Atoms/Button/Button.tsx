import React from "react";
import { ButtonContainer } from "./ButtonStyle";

export interface ButtonProps {
  children: React.ReactNode;
  onClick: any;
}

const Button: React.FC<ButtonProps> = ({ children, onClick }) => {
  return <ButtonContainer onClick={onClick}>{children}</ButtonContainer>;
};

export default Button;
