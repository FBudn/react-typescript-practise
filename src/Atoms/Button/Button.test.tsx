import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Button, { ButtonProps } from "./Button";
import "jest-styled-components";

const mockOnButtonClick = jest.fn();
const props: ButtonProps = {
  onClick: mockOnButtonClick,
  testId: `test-button-id`,
  children: `Button Test`,
};
// tekst osobny test zrobiÄ‡
test(`Should render Button Component`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeInTheDocument();
});

test(`Should match snapshot`, () => {
  const ButtonElement = TestRenderer.create(<Button {...props} />).toJSON();
  expect(ButtonElement).toMatchSnapshot();
});

test(`Should have styles and render`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeInTheDocument();
  expect(ButtonElement).toHaveStyle({
    display: "flex",
    width: "100%",
    height: "40px",
    borderRadius: "5px",
    border: "none",
    justifyContent: "center",
    alignItems: "center",
    fontWeight: "bold",
    color: "white",
    cursor: "pointer",
  });
});

test(`Should render with text`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeInTheDocument();
  expect(ButtonElement).toHaveTextContent(`Button Test`);
});

test(`Should render button and handle onClick`, () => {
  render(<Button {...props} />);
  const ButtonElement = screen.getByTestId(`test-button-id`);
  expect(ButtonElement).toBeInTheDocument();
  expect(mockOnButtonClick).toBeCalledTimes(0);
  fireEvent.click(ButtonElement);
  expect(mockOnButtonClick).toBeCalledTimes(1);
  mockOnButtonClick(`Button Test 1`);
  expect(mockOnButtonClick).toBeCalledWith(`Button Test 1`);
});
