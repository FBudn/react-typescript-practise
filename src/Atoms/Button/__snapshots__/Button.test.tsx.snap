// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 40px;
  border-radius: 5px;
  border: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  color: white;
  cursor: pointer;
  background: rgb(236 72 153);
  box-shadow: 0 4px 6px -1px rgb(0 0 0 / 0.1),0 2px 4px -2px rgb(0 0 0 / 0.1);
  --tw-shadow-color: #94a3b8;
  margin: 6px 0px 6px 0px;
}

<button
  className="c0"
  data-testid="test-button-id"
  onClick={[MockFunction]}
>
  Button Test
</button>
`;
